File: core/src/main/java/com/garden/back/notification/NotificationApplication.java
package com.garden.back.notification;

import com.garden.back.notification.domain.Notification;
import com.garden.back.notification.domain.NotificationType;
import com.garden.back.notification.service.EmailJavaMailNotificationService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
public class NotificationApplication {

    private final EmailJavaMailNotificationService javaMailService;

    public void send(Notification notification, NotificationType type) {
        switch (type) {
            case EMAIL -> {
                javaMailService.send(notification);
            }

            case SMS -> {
                throw new IllegalStateException("SMS Notification Not Yet Implemented");
            }

            case SMARTPHONE_PUSH -> {
                throw new IllegalStateException("Smartphone Push Notification Not Yet Implemented");
            }
        }
    }
}

File: core/src/main/java/com/garden/back/notification/NotificationApplication.kt
package com.garden.back.notification

import com.garden.back.notification.domain.Notification
import com.garden.back.notification.domain.NotificationType
import com.garden.back.notification.service.EmailJavaNotificationService
import com.garden.back.notification.service.SlackNotificationService
import org.springframework.stereotype.Component

@Component
class NotificationApplication(
    private val javaMailService: EmailJavaNotificationService,
    private val slackService: SlackNotificationService,
) {
    fun send(notification: Notification, type: NotificationType) {
        when (type) {
            NotificationType.EMAIL -> javaMailService.send(notification)
            NotificationType.SMS -> TODO("SMS Notification Not Yet Implemented")
            NotificationType.SMARTPHONE_PUSH -> TODO("Smartphone Push Notification Not Yet Implemented")
            NotificationType.SLACK -> slackService.send(notification)
        }
    }
}
